src/source/Ice/IceAgent.c

```diff
@@ -1193,6 +1224,15 @@ STATUS iceAgentSendStunPacket(PStunPacket pStunPacket, PBYTE password, UINT32 pa
             DLOGD("mark candidate pair %s_%s as failed", pIceCandidatePair->local->id, pIceCandidatePair->remote->id);
             pIceCandidatePair->state = ICE_CANDIDATE_PAIR_STATE_FAILED;
         }
+    } else {
+        CHK_STATUS(findIceCandidatePairWithLocalSocketConnectionAndRemoteAddr(pIceAgent, pLocalCandidate->pSocketConnection, pDestAddr, TRUE,
+                                                                              &pIceCandidatePair));
+        if (pIceCandidatePair == pIceAgent->pDataSendingIceCandidatePair && pIceCandidatePair != NULL) {
+            if (pIceAgent->pDataSendingIceCandidatePair->firstStunRequest) {
```

Comment:
Combine the two ifs together In this case the equivalent boolean would be

if (pIceCandidatePair == pIceAgent->pDataSendingIceCandidatePair && pIceCandidatePair != NULL && pIceAgent->pDataSendingIceCandidatePair->firstStunRequest) {

Reply:
Done

Reply:
Done